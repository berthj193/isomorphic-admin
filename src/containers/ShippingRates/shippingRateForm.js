import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import Form, { FormDivider, FormSubmit } from '../../components/uielements/form';
import Input from '../../components/uielements/input';
import Switch from '../../components/uielements/switch';
import CurrencySelect from '../../components/enum/currencySelect';
import CountrySelect from '../../components/enum/countrySelect';
import withReduxForm from '../../components/formWrapper/withReduxForm';
import WarehouseAutocomplete from '../../components/autocomplete/warehouse';

import { required, number, autocompleteIdRequired } from '../../helpers/formValidation';

export class NewShippingRateForm extends React.Component {
  static propTypes = {
    items: PropTypes.string,
    handleSubmit: PropTypes.func,
    warehouses: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
    })),
  };

  static defaultProps = {
    warehouses: [],
  };

  render() {
    return (
      <Form onSubmit={this.props.handleSubmit}>
        <Field label="Name" name="name" component={Input} validate={[required]} required />
        <Field
          label="Warehouse"
          name="warehouse_id"
          component={WarehouseAutocomplete}
          validate={[autocompleteIdRequired]}
          required
        />

        <FormDivider />
        <Field label="Currency" name="currency" component={CurrencySelect} validate={[required]} required />
        <Field label="Shipping price" name="price_pence" component={Input} validate={[required, number]} required />

        <FormDivider />
        <Field label="Country" name="country" placeholder="United Kingdom" component={CountrySelect} validate={[required]} required />
        <Field label="Shipping code" name="shipping_code" placeholder="NW1 5LS" component={Input} validate={[required]} required />

        <FormDivider />
        <Field label="Premium" name="premium" component={Switch} />
        <Field label="Live" name="live" component={Switch} />

        <FormSubmit />
      </Form>
    );
  }
}

const form = withReduxForm({
  form: 'shippingRateForm',
  enableReinitialize: true,
  initialValues: {
    live: true,
    premium: false,
  },
})(NewShippingRateForm);

export default form;
