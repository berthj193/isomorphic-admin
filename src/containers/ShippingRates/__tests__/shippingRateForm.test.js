import React from 'react';
import { NewShippingRateForm } from '../shippingRateForm';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<ShippingRateForm />', () => {
  const getComponent = (props = {}) => shallow(<NewShippingRateForm {...props} />);

  it('should render properly', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });
});
