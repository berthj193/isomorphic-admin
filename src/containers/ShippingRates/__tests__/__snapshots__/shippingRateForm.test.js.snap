// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingRateForm /> should render properly 1`] = `
<Form
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
  prefixCls="ant-form"
>
  <Field
    component={[Function]}
    label="Name"
    name="name"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="Warehouse"
    name="warehouse_id"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <FormDivider />
  <Field
    component={[Function]}
    label="Currency"
    name="currency"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="Shipping price"
    name="price_pence"
    required={true}
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
  <FormDivider />
  <Field
    component={[Function]}
    label="Country"
    name="country"
    placeholder="United Kingdom"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="Shipping code"
    name="shipping_code"
    placeholder="NW1 5LS"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <FormDivider />
  <Field
    component={[Function]}
    label="Premium"
    name="premium"
  />
  <Field
    component={[Function]}
    label="Live"
    name="live"
  />
  <FormSubmit />
</Form>
`;
